lib/wanda/executions/evaluation.ex

```diff
@@ -157,12 +157,25 @@ defmodule Wanda.Executions.Evaluation do
   end
 
   defp eval_expectation(
-         %Expectation{name: name, type: type, expression: expression},
+         %Expectation{
+           name: name,
+           type: type,
+           expression: expression,
+           failure_message: failure_message
+         },
          evaluation_scope
        ) do
     case Rhai.eval(expression, evaluation_scope) do
       {:ok, return_value} ->
-        %ExpectationEvaluation{name: name, type: type, return_value: return_value}
+        maybe_add_failure_message(
```

Comment:
I think this concept is wrong.
We shouldn't add the failure message to the expectation evaluation (which is related to the agent).
The failure message should go in the `ExpectationResult` object.
Otherwise, you cannot even know if a `expect_same` case failed, because the `return_value` is not a `true/false` thing.